### PR TITLE
Feat/shibe command

### PR DESCRIPTION
**Pull request by dmarticus**
this is my attempt at implementing https://github.com/WarwickTabletop/tablebot/issues/49.  Still needs some local testing but it seems reasonable, the API response is super basic.

### CODE REVIEW

File: src/Tablebot/Plugins/Shibe.hs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    plug,
+  )
+
+-- | @ShibeAPI@ is the basic data type for the JSON object that the Shibe API returns
+data ShibeAPI = ShibeAPI
+  { url :: !Text
+  }
+  deriving (Show, Generic)
+
+instance FromJSON ShibeAPI
```
Comment (finnbar) [inline src/Tablebot/Plugins/Shibe.hs:38]:
```suggestion
type ShibeAPI = Text
```
Shibe only returns a raw URL, while the `url :: !Text` pattern is for a key-value pair `url: ...`.

File: src/Tablebot/Plugins/Shibe.hs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  where
+    eitherHead [] = Left "Empty list"
+    eitherHead (x : _) = Right x
+
+-- | @getShibe@ is a helper function that turns the Either of @getShibeAPI@
+-- into either an error message or the url of the shibe image.
+getShibe :: Either String ShibeAPI -> Text
+getShibe esc = case esc of
+  (Left r) -> "no shibe today, sorry :(. (error is `" <> pack r <> "`)"
+  (Right r) -> url r
```
Comment (finnbar) [inline src/Tablebot/Plugins/Shibe.hs:70]: with the above `ShibeAPI` change, this just becomes
```suggestion
  (Right r) -> r
```

**Review by finnbar — CHANGES_REQUESTED** [review]
Hi! Thank you for contributing your code to our project, and sorry it took me so long to get back to you. (This is very much a small hobby project, hence why I'm kind of surprised you found us to be honest!)

The Shibe API has a slightly different format than the one you present in the `ShibeAPI` type, so doesn't currently work. I've attached changes that should fix that.

The other quick fix that needs doing (which I would have presented as changes, but it won't let me suggest changes to files that were unaffected by the commit) is that the plugin needs to actually be included in the bot. In `Tablebot.Plugins`, you'll need to import `shibePlugin` and add it to the list of plugins present in that file. I'll update the documentation at some point to make that clearer, because it's easy to miss.

Thank you again for your contribution!

**Comment by dmarticus** [comment]
Thanks checking this out @finnbar!  I found this because the project was featured in [Haskell Weekly](https://haskellweekly.news/issue/290.html), and I figured it was a pretty context-agnostic task :)  I just committed your suggestions and pushed up the change to include this plugin in the `Tablebot.Plugins` module!  

Lmk if there's anything else I need to do!

**Comment by finnbar** [comment]
Ah, thank you for doing that @dmarticus - although I think you've missed one of the suggestions which is why the build's failing right now. Otherwise you've already done the ormolu pass so it'll be good to go!

I'm a little surprised it was in Haskell Weekly just because I've made literally zero effort advertising this project, so it's a bit of a pleasant surprise that it was picked up by them! But anyway, all's good there.

**Pull request merged**